### PR TITLE
Patient Source

### PR DESCRIPTION
**Pull request by ljades**
Includes Patient Source and Test Cases

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] Internal ticket for this PR: https://jira.mitre.org/browse/TACO-6
- [x] Internal ticket links to this PR
- [x] Code diff has been done and been reviewed
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass

**Reviewer 1:**

Name: @holmesie 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

**Reviewer 2:**

Name: @okeefm
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

### CODE REVIEW

**Review by holmesie — CHANGES_REQUESTED** [review]
Let's get this repo to have `cqm-models` as a dependency (after we browserify that), and use browserify here as well. This will makes things work for Bonnie.

EDIT: My bad, looks like `cqm-models` is already a dependency. Also, on further discussion, it makes more sense to add browserify as a seperate PR from this.

**Comment by okeefm** [comment]
@holmesie Are you good with this PR, if we browserify in a future PR?

**Comment by holmesie** [comment]
@okeefm yeah, that's fine, but we will have to do it before we start experimenting with this and bonnie

**Comment by okeefm** [comment]
@holmesie if you're good, can you approve?

File: package.json [hunk 1]
```diff
@@ -2,10 +2,10 @@
   "name": "js-ecqm-engine",
   "version": "0.0.1",
   "description": "",
-  "main": "lib/executor.js",
+  "main": "lib/index.js",
   "scripts": {
     "test": "jasmine",
-    "lint": "eslint 'lib/*.js' 'lib/**/*.js'"
+    "lint": "eslint 'lib/*.js' 'lib/**/*.js' 'spec/*.js' 'spec/**/.js'"
```
Comment (okeefm) [inline package.json]: Does `spec/**/.js` actually get all the JS files? I feel like it should look more like `spec/**/*.js` to actually lint those files.

Comment (ljades) [reply package.json]: Pretty sure that's a typo. Changing it now.

**Review by okeefm — APPROVED** [review]
Assuming the API (things like `current()` and `nextPatient()`) make sense to the Bonnie team, who has more familiarity with the cql4browsers API, this looks good to me.

**Review by holmesie — APPROVED** [review]
